### PR TITLE
Chore/reorder imports

### LINKED ISSUE #3539: [eng quality] standardize import order
Proposing that we update import order such that each file uses three groupings:
- std/alloc/core
- external crate import
- imports from inside crate (self, super, crate)

### PR DESCRIPTION
**Pull request by pavitthrap**
### Description
- This PR standardizes the import order for all files such that they are in a maximum of three groups: 
1. A group with std/alloc/core imports 
2. A group with external crate imports 
3. A group for internal imports: self, super, and crate. 

- It updates the CI and checks for the correct import ordering + grouping. 
- It updates the location of CONTRIBUTING.md file (contents in CONTRIBUTORS.md were moved to docs/CONTRIBUTING.md). 

I used a rustfmt flag in order to do this reformatting. Here is the command:  `cargo fmt --all -- --config group_imports=StdExternalCrate`. To run just the check, you can run `cargo fmt --all -- --check --config group_imports=StdExternalCrate`. 

### Applicable issues
- fixes #3539 

### Checklist
- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Review by jcnelson — APPROVED** [review]

**Comment by jcnelson** [comment]
Looks like there needs to be at least one more rebase :/

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+- Use markdown to format comments.
+
+- Use the triple forward slash "///" for modules, structs, enums, traits and functions. Use double forward slash "//" for comments on individual lines of code.
+
+- Start with a high-level description of the function, adding more sentences with details if necessary.
+
+- When documenting panics, errors, or other conceptual sections, introduce a Markdown section with a single `#`, e.g.:
+
+- ```
```
Comment (obycode) [inline docs/CONTRIBUTING.md:218]: There is an extraneous `-` here which messes up the formatting for the rest of the [doc](https://github.com/stacks-network/stacks-blockchain/blob/972f061bb5a8dc0ccb200c648d71286a7eabf066/docs/CONTRIBUTING.md).

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+    PRs must include test coverage. However, if your PR includes large tests or tests which cannot run in parallel
+    (which is the default operation of the `cargo test` command), these tests should be decorated with `#[ignore]`.
+    If you add `#[ignore]` tests, you should add your branch to the filters for the `all_tests` job in our circle.yml
+    (or if you are working on net code or marf code, your branch should be named such that it matches the existing
+    filters there).
+
+    A test should be marked `#[ignore]` if:
+
+    1. It does not _always_ pass `cargo test` in a vanilla environment (i.e., it does not need to run with `--test-threads 1`).
+    2. Or, it runs for over a minute via a normal `cargo test` execution (the `cargo test` command will warn if this is not the case).
```
Comment (obycode) [inline docs/CONTRIBUTING.md:371]: This sub-section should not be under the Comments section.

**Review by obycode — CHANGES_REQUESTED** [review]
I just noticed a few issues with the merged contributing guide.

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (pavitthrap) [reply docs/CONTRIBUTING.md:218]: the indent had to be fixed here!

**Review by obycode — APPROVED** [review]
lgtm!

**Pull request merged**